merlin/spec/specification.py

```diff
@@ -268,7 +285,7 @@ def _process_dict_or_list(self, obj, string, key_stack, lvl, tab):
         list_offset = 2 * " "
         if isinstance(obj, list):
             n = len(obj)
-            use_hyphens = key_stack[-1] in ["paths", "sources", "git", "study"]
+            use_hyphens = key_stack[-1] in ["paths", "sources", "git", "study"] or key_stack[0] in ["user"]
```

Comment:
What is this change doing?

Reply:
Without this line, the pretty string version of the `user` block was incorrect when it contains a list- formatting with square brackets instead of new lines. This solves that problem. I do wonder if there is a more robust way of treating the `user` block as generic yaml and formatting accordingly